test/cmdlineTests/require_with_error_ir/output

```diff
@@ -0,0 +1,277 @@
+
+            }
+
+            function revert_error_42b3090547df1d2001c96683413b8cf91c1b902ef5e3cb8d9f6f304cf7446f74() {
+                revert(0, 0)
+            }
+
+            function cleanup_t_rational_1_by_1(value) -> cleaned {
+                cleaned := value
+            }
+
+            function cleanup_t_uint256(value) -> cleaned {
+                cleaned := value
+            }
+
+            function identity(value) -> ret {
+                ret := value
+            }
+
+            function convert_t_rational_1_by_1_to_t_uint256(value) -> converted {
+                converted := cleanup_t_uint256(identity(cleanup_t_rational_1_by_1(value)))
+            }
+
+            function abi_encode_t_rational_1_by_1_to_t_uint256_fromStack(value, pos) {
+                mstore(pos, convert_t_rational_1_by_1_to_t_uint256(value))
+            }
+
+            function array_storeLengthForEncoding_t_string_memory_ptr_fromStack(pos, length) -> updated_pos {
+                mstore(pos, length)
+                updated_pos := add(pos, 0x20)
+            }
+
+            function store_literal_in_memory_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720(memPtr) {
+
+                mstore(add(memPtr, 0), "two")
+
+            }
+
+            function abi_encode_t_stringliteral_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720_to_t_string_memory_ptr_fromStack(pos) -> end {
+                pos := array_storeLengthForEncoding_t_string_memory_ptr_fromStack(pos, 3)
+                store_literal_in_memory_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720(pos)
+                end := add(pos, 32)
+            }
+
+            function abi_encode_tuple_t_rational_1_by_1_t_stringliteral_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720__to_t_uint256_t_string_memory_ptr__fromStack(headStart , value0) -> tail {
+                tail := add(headStart, 64)
+
+                abi_encode_t_rational_1_by_1_to_t_uint256_fromStack(value0,  add(headStart, 0))
+
+                mstore(add(headStart, 32), sub(tail, headStart))
+                tail := abi_encode_t_stringliteral_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720_to_t_string_memory_ptr_fromStack( tail)
+
+            }
+
+            function require_helper_t_error_7_t_rational_1_by_1_t_stringliteral_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720(condition , expr_15) {
+                if iszero(condition) {
+                    let memPtr := allocate_unbounded()
+                    mstore(memPtr, 0x97ea5a2f00000000000000000000000000000000000000000000000000000000)
+                    let end := abi_encode_tuple_t_rational_1_by_1_t_stringliteral_332c39dcd398ea34a48b871898d589f55fc4c7bce00562fb670c972e7e1b0720__to_t_uint256_t_string_memory_ptr__fromStack(add(memPtr, 4) , expr_15)
+                    revert(memPtr, sub(end, memPtr))
+                }
+            }
```

Comment:
I see that we're going to generate a separate function not only for different string literals (where it makes sense, because they're not passed via variable) but also for different integer literals, where it's actually not necessary, because we have the parameter for that.

Not something to address in this PR since I see that ABI encoding functions already do the same thing, but I wonder if it it isn't something that would potentially improve optimization a little when eliminated. I guess [`EquivalentFunctionCombiner`](https://docs.soliditylang.org/en/develop/internals/optimizer.html#equivalent-function-combiner) might be able to eliminate that given enough iterations, or such functions might be inlined, but I'm a little concerned whether the new one will still be able to deal with that given that it has much less repetition. I think I'll need to pay attention to this with the new sequence.